barretenberg/cpp/src/barretenberg/transcript/transcript.hpp

```diff
@@ -71,6 +71,24 @@ struct NativeTranscriptParams {
     {
         return bb::field_conversion::convert_challenge<T>(challenge);
     }
+    /**
+     * @brief Split a challenge field element into two half-width challenges
+     * @details `lo` is 128 bits and `hi` is 126 bits.
+     * This should provide significantly more than our security parameter bound: 100 bits
+     *
+     * @param challenge
+     * @return std::array<Fr, 2>
+     */
+    static inline std::array<Fr, 2> split_challenge(const Fr& challenge)
+    {
+        static constexpr size_t LO_BITS = plookup::FixedBaseParams::BITS_PER_LO_SCALAR;
```

Comment:
weird to use plookup::FixedBaseParams::BITS_PER_LO_SCALAR, maybe we can move this constant somewhere else/rename it?

Reply:
also, shouldn't we try to generate 2 127 bit challenges as opposed to a 128 and a 126 bit challenge?

Reply:
yeah this could do with a renaming

Reply:
> also, shouldn't we try to generate 2 127 bit challenges as opposed to a 128 and a 126 bit challenge?

the extra bit of security isn't significant as the curve we use has 100-110 bits of security. I felt having a simple "cut this into 128-bit chunks" heuristic would be simpler to reason about.

also...splitting into 128/126 bits is 1 gate cheaper. `create_range_constraint` can cut 126 bits into 9 14-bit slices perfectly, which requires 3 gates to accumulate. 127/128 bits require 10 slices which needs 4 gates.

Reply:
> weird to use plookup::FixedBaseParams::BITS_PER_LO_SCALAR, maybe we can move this constant somewhere else/rename it?

I moved the definition into `field::Params` and its naming reflects why it's 128 (at least for `cycle_scalar`, it's to do with how large the scalar value can be when splitting a field element into two half-width slices according to the endormorphism trick)